hazelcast-spring/src/main/resources/hazelcast-spring-5.1.xsd

```diff
@@ -4712,6 +4713,51 @@
         </xs:attribute>
     </xs:complexType>
 
+    <xs:complexType name="dynamic-configuration">
+        <xs:annotation>
+            <xs:documentation>
+                Configuration settings related to dynamic configuration feature.
+            </xs:documentation>
+        </xs:annotation>
+        <xs:all>
+            <xs:element name="persistence-enabled" type="xs:boolean">
+                <xs:annotation>
+                    <xs:documentation>
+                        True if dynamic configuration persistence is enabled.
+                    </xs:documentation>
+                </xs:annotation>
+            </xs:element>
+            <xs:element name="persistence-file" type="xs:string" minOccurs="0">
+                <xs:annotation>
+                    <xs:documentation>
+                        Path for dynamic configuration persistence file.
+                        Dynamic changes will be persisted into this file.
+                        Don't forget to import this file in the root configuration file.
```

Comment:
How is this supposed to work?

In EE it persists the whole config, right? So it will persist config, which includes itself, can we handle that?

Reply:
It doesn't persist the whole config, it only persists newly added parts. But it will extend it if there isn't any exterior tags. For example you added a new map config, and this file is an empty file. It will add root tags as well and only persist that.

Reply:
https://github.com/hazelcast/hazelcast/pull/20170#issuecomment-1008868807